php/WP_CLI/Dispatcher/CommandFactory.php

```diff
@@ -127,6 +129,24 @@ private static function create_composite_command( $parent, $name, $callable ) {
 	}
 
 	/**
+	 * Create a new command namespace instance.
+	 *
+	 * @param mixed $parent The new namespace's parent Root or Composite command.
+	 * @param string $name Represents how the command should be invoked
+	 * @param mixed $callable
+	 */
+	private static function create_namespace( $parent, $name, $callable ) {
+		$reflection = new \ReflectionClass( $callable );
```

Comment:
Should we enforce any specific characteristics of the class (e.g. no defined methods, must extend some base class, etc.)?

Also, what happens if this isn't a class? `WP_CLI::add_command()` accepts any callable.

Reply:
This is a private helper method. The `$callable` was already checked before to assert that it is a subclass of the namespace class.

Reply:
Ah :)